### ISSUE TITLE
Installer fails .NET precondition even if already installed Win 10 .net 4.6

### ISSUE THREAD

**Issue by sachpatel** [nunit/nunit#1500]
I am having issues installing NUnit 3.2.1 and even Nunit 3.2.0 on a new machine running Windows 10.

The issue is similar to #939 except this is for installing a fresh copy with no previous installation of NUnit on the machine.

I have checked the registry at the following:

- HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\policy\v2.0 - containing a key for 50727 with a value 50727-50727 and another key under the same location called '(Default)' with no value set.
- HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\policy\v4.0 - containing a key for 30319 with a value 30319-30319 and another key under the same location called '(Default)' with no value set.

For .NET installation above 4.5, the place to check what the version installed is at HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\1033

Under the above location I have the 'Version' key set to 4.6.01038

I am unsure of what solutions I can suggest but hopefully this gives some information on how somebody else could confirm or reproduce the issue.

**Comment by ChrisMaddock** [comment]
I tried this on a fresh machine recently after similar was reported in #1214 - but was able to install with no problem. All my registry keys match yours. Afraid I don't have anything more constructive to add - other that assuming there must be something non-standard going on.

**Comment by sachpatel** [comment]
I may be clutching at straws. I haven't used Wix in a long time.

I think the issue may be to the nunit/install/common/variables.wxi file that was last edited with the fix for #943 

The condition is: `<![CDATA[Installed OR FRAMEWORK20 OR FRAMEWORK40 OR MONODIRECTORY]]>`

I believe the FRAMEWORK20, FRAMEWORK40 properties hold the registry value not a boolean value the condition is able to interpret correctly.

The documentation at the following link shows it may be better to add the `result="exists"` attribute to the RegistrySearch elements used in the condition of the variables.wxi file.

http://wixtoolset.org/documentation/manual/v3/xsd/util/registrysearch.html

Since most people are able to install the latest version of NUnit without issues I may just give the above a go when I have a few moments and update this issue/submit PR.

**Comment by ChrisMaddock** [comment]
@sachpatel - I initially thought similar, however this wix example is written as NUnit's - so I presumed it did some sort of bitwise OR...couldn't find anything defining behaviour there! You could be write, and the sample could be in error, but then I'm not sure why that would differ per machine.
http://wixtoolset.org/documentation/manual/v3/howtos/files_and_registry/read_a_registry_entry.html

If you're able to build and test an installer on you machine, that would be great - I've tried on a few machines now, and have been unable to reproduce. 

The only other thing that stands out to me, is that in my (very limited) wix experience, we've always used the build in variables in `WixNetFxExtension` to check for .NET installation. I can't see how what we have _wouldn't_ work...but did wonder if there might be some edge cases wrapped up in the extension we're not aware of.

**Comment by sachpatel** [comment]
There is definitely something odd going on. I forked the repo, pulled it down and built the projects. Then used `build.cmd -t Package` to produce the msi. The resulting msi when executed through double clicking in Windows explorer throws up the "NUnit requires .NET 2.0, .NET 4.0 or Mono to be installed as a prerequisite." error.

Trying to run the msi installation from the elevated command line with `msiexec /i NUnit-3.3.0.msi` throws up:
![image](https://cloud.githubusercontent.com/assets/10399742/15358457/798d18e4-1cfc-11e6-8882-28670ad94e84.png)

msiexec does not like msi files with periods or dashes. So I renamed the msi to NUnit3.msi and run `msiexec /i NUnit3.msi` and I was able to successfully install NUnit 3.3.0

This has caused a lot of confusion and I am no closer to finding the reason why this works but double clicking the msi and successfully installing doesn't.

I went the msiexec route because I wanted some sort of logging from the installation and was running `msiexec /i NUnit3.msi /l*v Package.log` before I found that the installation succeeded.

**Comment by ChrisMaddock** [comment]
@sachpatel - Do all applications have permission to read the relevant keys? (right click > permissions, if you're looking at it with regedit.msc)

I wasn't aware there could be per-key read permissions.

**Comment by sachpatel** [comment]
When I go to the permissions of the policy folder under regedit and even the v2.0 and v4.0 sub folders the 'ALL APPLICATION PACKAGES' group has read permissions.

Thank you for your help. I think this is an issue that will be hard to get to the bottom of.

**Comment by appel1** [comment]
I've never heard of msiexec having issues with period or dash characters before. I've worked on products deployed using msis and patches with such characters in the file name for years and have never had any issues related to the file name. Something else must be going on on your machine. Perhaps some group policy or security software is interfering?

**Comment by sachpatel** [comment]
@appel1 I found changing the filename of the msi by removing the periods and dashes to have allowed me to successfully run the installation launched from msiexec without the error I was seeing otherwise. It may just be a problem I experienced as it does sound odd. I did not mean to infer that msiexec has a known bug/problem with filenames containing periods or dashes.

I have been able to install NUnit successfully. I still do not know why I was experiencing the error and have not spent any more time in finding the issue. It may well be to do with policy/permission settings.
